### PR TITLE
feat: add textarea handling on textfield codemod

### PR DESCRIPTION
**Pull request by massao**
# Purpose of PR

Add handling of textfields that had a `textarea` prop on the codemod.

### CODE REVIEW

File: packages/forma-36-codemod/transforms/__testfixtures__/v4/v4-text-field.output.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { FormControl, TextInput, Flex } from "@contentful/f36-components";
+import { FormControl, TextInput, Flex, TextArea } from "@contentful/f36-components";
```
Comment (denkristoffer) [inline packages/forma-36-codemod/transforms/__testfixtures__/v4/v4-text-field.output.js:1]:
```suggestion
import { FormControl, TextInput, Flex, Textarea } from "@contentful/f36-components";
```

File: packages/forma-36-codemod/transforms/__testfixtures__/v4/v4-text-field.output.js [hunk 2]
```diff
@@ -124,3 +124,8 @@ const validationMessageCondition = 0;
   <TextInput name="conditional-validation" onChange={() => {}} />
   {!(validationMessageCondition > 0) && <FormControl.ValidationMessage>Some validation message</FormControl.ValidationMessage>}
 </FormControl>;
+
+<FormControl id="textarea">
+  <FormControl.Label>Text area</FormControl.Label>
+  <TextArea name="textarea" value="some value" rows={5} />
```
Comment (denkristoffer) [inline packages/forma-36-codemod/transforms/__testfixtures__/v4/v4-text-field.output.js:130]:
```suggestion
  <Textarea name="textarea" value="some value" rows={5} />
```

File: packages/forma-36-codemod/transforms/v4-text-field.js [hunk 3]
```diff
@@ -46,6 +47,15 @@ function textFieldCodemod(file, api) {
       );
       const formControlProps = [...commonProps, id, isRequired];
 
+      // Check if is textarea
+      const isTextarea = hasProperty(attributes, { propertyName: 'textarea' });
+      if (isTextarea && !componentsToImport.includes('TextArea')) {
+        componentsToImport.push('TextArea');
```
Comment (denkristoffer) [inline packages/forma-36-codemod/transforms/v4-text-field.js:53]:
```suggestion
        componentsToImport.push('Textarea');
```

File: packages/forma-36-codemod/transforms/v4-text-field.js [hunk 4]
```diff
@@ -46,6 +47,15 @@ function textFieldCodemod(file, api) {
       );
       const formControlProps = [...commonProps, id, isRequired];
 
+      // Check if is textarea
+      const isTextarea = hasProperty(attributes, { propertyName: 'textarea' });
+      if (isTextarea && !componentsToImport.includes('TextArea')) {
```
Comment (denkristoffer) [inline packages/forma-36-codemod/transforms/v4-text-field.js:52]:
```suggestion
      if (isTextarea && !componentsToImport.includes('Textarea')) {
```

File: packages/forma-36-codemod/transforms/v4-text-field.js [hunk 5]
```diff
@@ -95,7 +105,7 @@ function textFieldCodemod(file, api) {
 
       const TextInput = createComponent({
         j,
-        componentName: 'TextInput',
+        componentName: isTextarea ? 'TextArea' : 'TextInput',
```
Comment (denkristoffer) [inline packages/forma-36-codemod/transforms/v4-text-field.js:108]:
```suggestion
        componentName: isTextarea ? 'Textarea' : 'TextInput',
```

**Review by denkristoffer — CHANGES_REQUESTED** [review]

**Comment by denkristoffer** [comment]
This makes me think we might want to test that the code from our codemods can be compiled, so we don't just match input to output, but also try to build it.

**Review by denkristoffer — APPROVED** [review]

**Review by Lelith — APPROVED** [review]
Looks good to me!

**Pull request merged**